calendar-dav/src/main/java/com/linagora/calendar/dav/CalDavClient.java

```diff
@@ -459,4 +464,76 @@ public Mono<Void> updateCalendarAcl(Username username, CalendarURL calendarURL,
                         """.formatted(response.status().code(), uri, errorBody))));
             });
     }
+
+    private byte[] buildPatchDelegationBodyRequest(List<Username> addOrUpdateAdmins, List<Username> revokeAdmins) {
+        try {
+            ObjectNode share = OBJECT_MAPPER.createObjectNode();
+            share.set("set", rightsToAddAsJson(addOrUpdateAdmins));
+            share.set("remove", rightsToRemoveAsJson(revokeAdmins));
+
+            ObjectNode body = OBJECT_MAPPER.createObjectNode().set("share", share);
+            return OBJECT_MAPPER.writeValueAsBytes(body);
+        } catch (JsonProcessingException e) {
+            throw new DavClientException("Failed to serialize JSON for patching read/write delegations", e);
+        }
+    }
+
+    private ArrayNode rightsToRemoveAsJson(List<Username> revokeAdmins) {
+        return revokeAdmins.stream()
+            .collect(OBJECT_MAPPER::createArrayNode,
+                (arrayNode, admin) -> arrayNode.add(OBJECT_MAPPER.createObjectNode()
+                    .put("dav:href", "mailto:" + admin.asString())),
+                ArrayNode::addAll);
+    }
+
+    private ArrayNode rightsToAddAsJson(List<Username> addAdmins) {
+        return addAdmins.stream()
+            .collect(OBJECT_MAPPER::createArrayNode,
+                (arrayNode, admin) -> arrayNode.add(OBJECT_MAPPER.createObjectNode()
+                    .put("dav:href", "mailto:" + admin.asString())
+                    .put("dav:read-write", true)),
+                ArrayNode::addAll);
+    }
```

Comment:
```suggestion
        return addAdmins.stream()
            .map(admin -> OBJECT_MAPPER.createObjectNode()
                    .put("dav:href", "mailto:" + admin.asString())
                    .put("dav:read-write", true)))
            .collect(OBJECT_MAPPER::createArrayNode,
                (arrayNode, adminNode) -> arrayNode.add(adminNode),
                ArrayNode::addAll);
    }
```

We can extract the map to even keep increasing readability.